src/Phinx/Migration/Manager.php

```diff
@@ -173,12 +181,36 @@ public function migrate($environment, $version = null)
                 break;
             }
 
-            if (!in_array($migration->getVersion(), $versions)) {
+            if (in_array($migration->getVersion(), $pendingVersions)) {
+                list($versions, $pendingVersions) = $this->waitOnPendingMigration($migration, $env);
+            }
+
+            if (! in_array($migration->getVersion(), $versions)) {
                 $this->executeMigration($environment, $migration, MigrationInterface::UP);
             }
         }
     }
-    
+
+    protected function waitOnPendingMigration(MigrationInterface $migration, Environment $env)
+    {
+        $timeWaited = 0;
+        $this->getOutput()->writeln('');
+        do {
+            if ($timeWaited % (static::POLL_DELAY * (int) static::POLL_MESSAGE_FREQUENCY) == 0) {
+                $this->getOutput()->writeln("<comment>Migration <info>{$migration->getVersion()}</info> in progress. Waiting ($timeWaited secs)...</comment>");
```

Comment:
Out put this at start of wait, without line break. On each loop, output a `.`, also without a line break. Output a line break after finishing looping.